pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1766,18 +1766,32 @@ private void startWorkerService(AuthenticationService authenticationService,
                                     AuthorizationService authorizationService)
             throws Exception {
         if (functionWorkerService.isPresent()) {
-            if (workerConfig.isUseTls() || brokerServiceUrl == null) {
+            if (isNotBlank(brokerServiceUrlTls)) {
```

Comment:
Do we also need to check `workerConfig.isUseTls()`? It has been deprecated but hasn't been removed yet. If there is an existing cluster that has `brokerServiceUrlTls` configured, but `useTls` is false. We should not use TLS in this case?

Reply:
I think we should use TLS in this case. because we check `tlsPort` not `useTls`

Reply:
Users should be able to control whether or not to use TLS themselves.

Reply:
yes.

Reply:
`brokerServiceUrlTls ` is also configured by the users.

Reply:
> So we want to break a case that doesn't seem very reasonable(They have configured brokerServiceUrlTls but isUseTls is false). Is that correct?

I think so. We have stated this behavior in the comment: https://github.com/apache/pulsar/blob/f7c0b3c49c9ad8c28d0b00aa30d727850eb8bc04/pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/worker/WorkerConfig.java#L458-L465.

But the problem is we didn't state it in the documentation.